Makefile

```
@@ -0,0 +1,24 @@
+PACKAGE_VERSION = $(shell cat VERSION)
+COV_CONFIG = ".coveragerc"
+SOURCE_FOLDER = "src"
+TESTS_FOLDER = "tests"
+TESTS_PARAMS = "-p no:cacheprovider"
+
+
+.PHONY: check
+check:
+	@echo "Checking code format"
+	@black --check $(SOURCE_FOLDER)
+	@black --check $(TESTS_FOLDER)
+
+
+tag-%:
+	@echo "Bumping and tagging version"
+	@$(eval PART := $*)
+	@bump2version --current-version $(PACKAGE_VERSION) $(PART)
```

`bump2version` is smart enough that you just need to tell it the part (`major`, `minor`, `patch`) that you want it to bump by. So this just removes `PACKAGE_VERSION` and simplifies the `bump2version` line.

I can't push to this repo, else I would just have added this commit (@Sinclert @lukasheinrich can you give @kratsg and me permissions?).
```suggestion
COV_CONFIG = ".coveragerc"
SOURCE_FOLDER = "src"
TESTS_FOLDER = "tests"
TESTS_PARAMS = "-p no:cacheprovider"


.PHONY: check
check:
	@echo "Checking code format"
	@black --check $(SOURCE_FOLDER)
	@black --check $(TESTS_FOLDER)


tag-%:
	@echo "Bumping and tagging version"
	@$(eval PART := $*)
	@bump2version $(PART)
```

---

I know.

I left it as a way of stating that the one that controls the package version, **is the `VERSION` file**. At the end, `bump2version` is just one of the options to manage versioning on a package, and one particular tools within the multiple ones a package usually gets integrated with (i.e. Sphinx, Docker...).

I would rather have a _tool-agnostic_, _language-agnostic_ file managing the version literal.

_P.S: The only reason I added to `.bumpversion.cfg` is because, apparently, it gets added every time bump2version is called._

---

> I left it as a way of stating that the one that controls the package version, is the `VERSION` file.

Sure but this has nothing to do with `bump2version` and the information is not at all needed in the Makefile. You can of course leave it in, but understand that it does nothing given that `bump2version` is also going to bump the `VERSION` file &mdash; it would be useless if it didn't. This doesn't invalidate `VERSION` in anyway, as that is what decides things at _runtime_ which is the thing you care about.

> The only reason I added to `.bumpversion.cfg` is because, apparently, it gets added every time bump2version is called.

No, it is a rather important file as it tells `bump2version` what to actually update for the whole project. c.f. [`pyhf`'s for an example of why this is so important](https://github.com/scikit-hep/pyhf/blob/master/.bumpversion.cfg).

---

It is true that the version bumping can be done without the `Makefile` variable, but I imagine it as a way of making explicit what otherwise would be implicit. Having the package version available, as a variable, in the `Makefile`, is not only for this particular `Makefile` rules, but for those that may come in the future.

It is the same as [adding a cron-job to the CI triggers](https://github.com/HEPData/hepdata-pyhf-extractor/pull/2#discussion_r508006559), even thought there are no **installation** requirements to worry about just yet. There will be, and the CI is ready for that.

---

Regarding my P.S. line, I misspelled, I wanted to say that _"I added **the version** to `.bumpversion.cfg`.."._